### PR TITLE
`dns::record::ptr::by_ip`: Permit `zone` parameter to be empty

### PR DESCRIPTION
**Pull request by jearls**
... to allow fully-qualified `host` values.

Also create basic spec tests for dns`dns::record::ptr::by_ip` type

### CODE REVIEW

**Comment by jearls** [comment]
This is an alternate proposal for #202

File: manifests/record/ptr/by_ip.pp [hunk 1]
```diff
@@ -102,10 +104,16 @@
...
+  } else {
+    $fqdn = $host
+  }
+
   dns::record::ptr { "${octet}.${reverse_zone}":
     host => $octet,
     zone => $reverse_zone,
     ttl  => $ttl ,
-    data => "${host}.${zone}"
+    data => $fqdn ,
```
Comment (solarkennedy) [inline manifests/record/ptr/by_ip.pp]: Can you remove all these ` ,` (extra spaces before the commas)
I'm not sure there is a specific style guide rule to that, but it just looks kinda different to me

File: manifests/record/ptr/by_ip.pp [hunk 2]
```diff
@@ -10,13 +10,16 @@
 # The ip address for the ptr record.  Defaults to the resource title.
 #
 # * `$host`
-# The (unqualified) host pointed to by this ptr record.
+# The host pointed to by this ptr record.  This must be fully qualified
```
Comment (solarkennedy) [inline manifests/record/ptr/by_ip.pp]: Can you clarify if there is supposed to be a period or not?

**Comment by jearls** [comment]
Okay, I fixed the comments for both `$host` and `$zone` to indicate *no* trailing `.`.

I also cleaned up the spaces-before-commas.  I use that in my own modules because, to me, it's much more readable.  Most of the time, I remember not to do that when I'm working on other people's code.  Sorry about that.

**Pull request merged**